bfdd/bfdd_cli.c

```diff
@@ -334,20 +348,27 @@ DEFPY(
 	"Configure peer echo interval\n"
 	"Configure peer echo interval value in milliseconds\n")
 {
+	char value[32];
+
+	snprintf(value, sizeof(value), "%ld", interval * 1000);
 	nb_cli_enqueue_change(vty, "./desired-echo-transmission-interval",
-			      NB_OP_MODIFY, interval_str);
+			      NB_OP_MODIFY, value);
+
 	return nb_cli_apply_changes(vty, NULL);
 }
 
 void bfd_cli_show_echo_interval(struct vty *vty, struct lyd_node *dnode,
 				bool show_defaults)
 {
+	uint32_t value;
+
 	if (show_defaults)
 		vty_out(vty, "  echo-interval %d\n",
 			BFD_DEF_REQ_MIN_ECHO);
-	else
-		vty_out(vty, "  echo-interval %s\n",
-			yang_dnode_get_string(dnode, NULL));
+	else {
+		value = yang_dnode_get_uint32(dnode, NULL);
+		vty_out(vty, "  echo-interval %" PRIu32 "\n", value / 1000);
+	}
```

Comment:
so echo-interval remains in milliseconds.right?
one more thought: I feel that the bfd topotests may have to be adapted because of the new settings added by default ( vrf, but also the default timer values expressed in microseconds)

Reply:
`echo-interval` was also converted to `microseconds`. Check the YANG model.

Reminder: the data structures have the timers store in microseconds. The only place where we need conversion is in the CLI input/output.

Reply:
I confirm also that the json values obtained through show bfd peers are expressed in milliseconds.
should not we clarify this : microseconds , and not milliseconds?